x/crosschain/keeper/verify_block_header.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func (k Keeper) VerifyProof(ctx sdk.Context, proof *common.Proof, hash string, txIndex int64, chainID int64) (ethtypes.Transaction, error) {
+	var txx ethtypes.Transaction
+	crosschainFlags, found := k.zetaObserverKeeper.GetCrosschainFlags(ctx)
+	if !found {
+		return txx, fmt.Errorf("crosschain flags not found")
+	}
+	if common.IsBitcoinChain(chainID) && !crosschainFlags.BlockHeaderVerificationFlags.IsBtcTypeChainEnabled {
+		return txx, fmt.Errorf("cannot verify proof for bitcoin chain %d", chainID)
+	}
+
+	if common.IsEVMChain(chainID) && !crosschainFlags.BlockHeaderVerificationFlags.IsEthTypeChainEnabled {
```

Comment:
I'm wondering, shouldn't be `IsEthereumChain`?
I may think the header verification is implemented for Ethereum but not general to all EVM chains

Reply:
I decided against implementing a chain-specific param, right now just to keep things a bit simple. 
This should account for both BNB and ETH